drivers/i2c/i2c_cc13xx_cc26xx.c

```diff
@@ -58,12 +58,18 @@ static int i2c_cc13xx_cc26xx_transmit(const struct device *dev,
 	const uint32_t base = get_dev_config(dev)->base;
 	struct i2c_cc13xx_cc26xx_data *data = get_dev_data(dev);
 
-	/* Sending address without data is not supported */
+	I2CMasterSlaveAddrSet(base, addr, false);
+
+	/* Sending address without data */
 	if (len == 0) {
-		return -EIO;
+
+		I2CMasterControl(base, I2C_MASTER_CMD_SINGLE_SEND);
+
+		k_sem_take(&data->complete, K_FOREVER);
+
+		return data->error == I2C_MASTER_ERR_NONE ? 0 : -EIO;
 	}
 
-	I2CMasterSlaveAddrSet(base, addr, false);
```

Comment:
Does this really work as expected?

As far as I can see, `I2C_MASTER_CMD_SINGLE_SEND` sends both address and data (written to `I2CMDR`); and we do not write to `I2CMDR` in this case, so we will be sending garbage data here:

![image](https://user-images.githubusercontent.com/11345418/134097546-040dcfba-e6de-41d2-be36-fc8050829a47.png)

Can you post an oscilloscope or logic analyser capture of the SCL/SDA waveforms and confirm that no data is sent here?

Reply:
Your trace matches what I see with a logic analyzer. It is disappointing that the peripheral doesn't seem to be able to send just the address without also pushing a data byte.

![image](https://user-images.githubusercontent.com/1348448/134100433-2b35b7b5-85ff-4291-8a05-b9f3439b7710.png)

Reply:
@leadercxn let's figure out why there is append a  data.

Reply:
I consult this document -- CC13x2, CC26x2 SimpleLink™ Wireless MCU Technical Reference Manual (Rev. D) , in chapter 25.3.5 , it introduces several models of I2C communication, but i can't see any modes that just send the address without I2C_MDR data.
In the chapter 23.5 , there are no any registers refer to how to just send the address. I try to modify the MCTRL register, but it doesn't work . so i don‘t think it's hardware I2C  supports this feature that  just send the address without I2C_MDR data.

![i2c_scan](https://user-images.githubusercontent.com/43691182/134462863-650c59c5-8cc2-4d46-8169-3ce784d4d1b4.png)

[CC13x2, CC26x2 SimpleLink™ Wireless MCU Technical Reference Manual](https://www.ti.com/lit/ug/swcu185d/swcu185d.pdf?ts=1632376017614&ref_url=https%253A%252F%252Fwww.ti.com%252Fproduct%252FCC1352P)

Reply:
@devanlai  @stephanosio  It seems that after some research, this additional data is justified.

Reply:
> It seems that after some research, this additional data is justified.

@Pillar1989 Could you elaborate?

Reply:
In chapter 23.5, which has many Master TRANSMIT modes.  But every mode needs append data after the i2c address. 

![image](https://user-images.githubusercontent.com/4081906/134496395-4970ff67-0407-481d-bb27-a17940a082f2.png)

Reply:
> In chapter 23.5, which has many Master TRANSMIT modes. But every mode needs append data after the i2c address.

@Pillar1989 That means that the original comment is correct and this change is invalid.

> `/* Sending address without data is not supported */`